### PR TITLE
Add DeleteFilesInRange related test

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
cover the case that DeleteFilesInRange leads to incomplete snapshots

## What are the type of the changes? (mandatory)
- Bug fix (change which fixes an issue)

## How has this PR been tested? (mandatory)
unit-test

## Does this PR affect documentation (docs) or release note? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

**Comment by Connor1996** [comment]
need merge https://github.com/pingcap/titan/pull/18 first

File: tests/integrations/config/test-custom.toml [hunk 1]
```diff
@@ -173,6 +173,7 @@ enabled = true
 dirname = "bar"
 disable-gc = false
 max-background-gc = 9
+purge-obsolete-files-period = 1s
```
Comment (MyonKeminta) [inline tests/integrations/config/test-custom.toml]: `"1s"`?

File: src/raftstore/store/region_snapshot.rs [hunk 2]
```diff
@@ -379,6 +389,40 @@ mod tests {
         )
     }
 
+    fn new_temp_titan_engine(path: &TempDir) -> Engines {
+        let mut cfg = TiKvConfig::default();
+        let cache = cfg.storage.block_cache.build_shared_cache();
+        cfg.rocksdb.titan.enabled = true;
+        cfg.rocksdb.titan.purge_obsolete_files_period = ReadableDuration::secs(1);
+        cfg.rocksdb.defaultcf.disable_auto_compactions = true;
+        cfg.rocksdb.defaultcf.dynamic_level_bytes = false;
```
Comment (yiwu-arbug) [inline src/raftstore/store/region_snapshot.rs]: why `dynamic_level_bytes=false` is needed? Add a comment here?

File: src/raftstore/store/region_snapshot.rs [hunk 3]
```diff
@@ -785,4 +902,95 @@ mod tests {
         res.sort();
         assert_eq!(res, test_data[1..3].to_vec());
     }
+
+    #[test]
+    fn test_delete_files_in_range() {
```
Comment (yiwu-arbug) [inline src/raftstore/store/region_snapshot.rs]: Add some inline comment the explain the test? Like at which point which blob file will get GC-ed, which key will be left in the DB if we hit the previous Titan bug, etc.

File: components/engine/src/iterable.rs [hunk 4]
```diff
@@ -189,6 +189,5 @@ where
             break;
         }
     }
-
-    it.status().map_err(From::from)
+    Ok(())
```
[the remark this answers is not in the document]
Comment (Connor1996) [reply components/engine/src/iterable.rs]: just for reproduce `default not found`

**Comment by Connor1996** [comment]
Add the test in https://github.com/tikv/tikv/pull/4966, so close it

**Pull request closed without merging**